src/Service/SF1514/SF1514.php

```diff
@@ -0,0 +1,398 @@
+
+        // ReplyTo.
+        $replyToId = '_' . $this->generateUuid();
+        $replyToElement = $this->getElement($xpath, '//wsa:ReplyTo');
+        $replyToElement->setAttribute('wsu:Id', $replyToId);
+
+        $this->handleReference($xpath, $replyToElement, $replyToId, 'reply_id');
+
+        // Timestamp.
+        $timestampId = 'TS-' . $this->generateUuid();
+        $timestampElement = $this->getElement($xpath, '//wsu:Timestamp');
+        $timestampElement->setAttribute('wsu:Id', $timestampId);
+
+        $this->getElement($xpath, 'wsu:Created', $timestampElement)->nodeValue = $this->getTimestamp();
+        $this->getElement($xpath, 'wsu:Expires', $timestampElement)->nodeValue = $this->getTimestamp(300);
+
+        $this->handleReference($xpath, $timestampElement, $timestampId, 'timestamp_id');
+
+        // BinarySecurityToken.
+        $certificateKeyContent = str_replace(["\r", "\n"], '', $cert);
+
+        $binarySecurityTokenId = 'X509-' . $this->generateUuid();
+        $binarySecurityTokenElement = $this->getElement($xpath, '//wsse:BinarySecurityToken');
+        $binarySecurityTokenElement->setAttribute('wsu:Id', $binarySecurityTokenId);
+        $binarySecurityTokenElement->nodeValue = $certificateKeyContent;
+
+        $this->handleReference($xpath, $binarySecurityTokenElement, $binarySecurityTokenId, 'security_token_id');
+
+        // Body.
+        $bodyId = '_' . $this->generateUuid();
+        $bodyElement = $this->getElement($xpath, '//soap:Body');
+        $bodyElement->setAttribute('wsu:Id', $bodyId);
+
+        $this->getElement($xpath, '//wsauth:Value')->nodeValue = $cvr;
+        $this->getElement($xpath, 'wsse:BinarySecurityToken', $this->getElement($xpath, '//wst:UseKey'))->nodeValue = $certificateKeyContent;
+
+        $this->handleReference($xpath, $bodyElement, $bodyId, 'body_id');
+
+        // KeyInfo.
+        $keyInfoId = 'KI-' . $this->generateUuid();
+        $keyInfoElement = $this->getElement($xpath, '//ds:KeyInfo');
+        $keyInfoElement->setAttribute('Id', $keyInfoId);
+
+        // Set final ids.
+        $this->getElement($xpath, '//wsse:Reference')->setAttribute('URI', '#' . $binarySecurityTokenId);
+        $this->getElement($xpath, '//wsse:SecurityTokenReference')->setAttribute('wsu:Id', 'STR-' . $this->generateUuid());
+
+        // Sign the request.
+        $signedInfoElement = $this->getElement($xpath, '//ds:SignedInfo');
+
+        $signedIntoElementCanonical = $signedInfoElement->C14N(true, false);
+
+        openssl_sign($signedIntoElementCanonical, $signatureValue, $privKey, 'sha256WithRSAEncryption');
+
+        $signatureValue = base64_encode($signatureValue);
+        $this->getElement($xpath, '//ds:SignatureValue')->nodeValue = $signatureValue;
+
+        return $dom->saveXML();
+    }
+
+    /**
+     * Handles reference by signing element.
+     */
+    private function handleReference(DOMXPath $xpath, \DOMElement $element, string $elementId, $baseId)
```

Comment:
Inconsistent use of classes in the global namespace. Use either `\DOMXPath $xpath, \DOMElement $element` or use `use` to import the classes.

Reply:
Fixed.